app/services/invoices/refresh_draft_service.rb

```diff
@@ -95,5 +91,25 @@ def invoice_credit_note_items
         .joins(:credit_note)
         .where(credit_note: {invoice_id: invoice.id})
     end
+
+    def tax_error?(error)
+      error&.code == 'tax_error'
+    end
+
+    def reset_invoice_values
+      invoice.credit_notes.each { |cn| cn.items.update_all(fee_id: nil) } # rubocop:disable Rails/SkipsModelValidations
+      invoice.fees.destroy_all
+      invoice_subscriptions.destroy_all
+      invoice.applied_taxes.destroy_all
+      invoice.error_details.discard_all
+
+      invoice.taxes_amount_cents = 0
+      invoice.total_amount_cents = 0
+      invoice.taxes_rate = 0
+      invoice.fees_amount_cents = 0
+      invoice.sub_total_excluding_taxes_amount_cents = 0
+      invoice.sub_total_including_taxes_amount_cents = 0
+      invoice.save
```

Comment:
Do we really need last 7 lines? This is extra DB call but all those fields will be populated with new values anyway, once the refresh is done

Reply:
yes, because if we have an invoice where everything is calculated, including taxes, then we were refreshing an invoice and got an error - we won't be able to recalculate taxes, and without this nullifying it will keep old values

Reply:
Hmm, this will work but not sure if this extra save is needed. Even if old values stay attached, we won't display it on UI I think (we display "-" for each attribute that can't be calculated. But ok, maybe can be zero if someone fetched it from the API

Reply:
by the way, what do you thing about having `nil`s instead of `0`s there?

Reply:
nah, sorry, forget about it - we have default 0s  🤦‍♀️